src/txreconciliation.h

```diff
@@ -0,0 +1,585 @@
+
+    /**
+     * When a reconciliation request is received, instead of responding to it right away,
+     * we schedule a response for later, so that a spy can’t monitor our reconciliation sets.
+     */
+    std::chrono::microseconds m_next_recon_respond{0};
+
+    /** Keep track of reconciliations with the peer. */
+    ReconciliationPhase m_incoming_recon{RECON_NONE};
+    ReconciliationPhase m_outgoing_recon{RECON_NONE};
+
+    /**
+     * Reconciliation sketches are computed over short transaction IDs.
+     * Short IDs are salted with a link-specific constant value.
+     */
+    uint32_t ComputeShortID(const uint256 wtxid) const
+    {
+        const uint64_t s = SipHashUint256(m_k0, m_k1, wtxid);
+        const uint32_t short_txid = 1 + (s & 0xFFFFFFFF);
+        return short_txid;
+    }
+
+    void ClearState()
+    {
+        m_local_short_id_mapping.clear();
+        // This is currently belt-and-suspenders, as the code should work even without these calls.
+        m_remote_sketch_snapshot.clear();
+        m_local_set_snapshot.clear();
+        m_capacity_snapshot = 0;
+    }
+
+    /**
+     * Reconciliation involves computing a space-efficient representation of transaction identifiers
+     * (a sketch). A sketch has a capacity meaning it allows reconciling at most a certain number
+     * of elements (see BIP-330). Considering whether we are going to send a sketch to a peer or use
+     * locally, we estimate the set difference.
+     */
+    Minisketch ComputeSketch(uint16_t capacity, bool use_snapshot=false)
+    {
+        Minisketch sketch;
+        std::set<uint256> working_set;
+        if (use_snapshot) {
+            working_set = m_local_set_snapshot;
+        } else {
+            working_set = m_local_set;
+            m_capacity_snapshot = capacity;
+        }
+        // Avoid serializing/sending an empty sketch.
+        if (working_set.size() == 0 || capacity == 0) return sketch;
+
+        std::vector<uint32_t> short_ids;
+        for (const auto& wtxid: working_set) {
+            uint32_t short_txid = ComputeShortID(wtxid);
+            short_ids.push_back(short_txid);
+            m_local_short_id_mapping.emplace(short_txid, wtxid);
+        }
+
+        capacity = std::min(capacity, MAX_SKETCH_CAPACITY);
+        sketch = Minisketch(RECON_FIELD_SIZE, 0, capacity);
+        if (sketch) {
+            for (const uint32_t short_id: short_ids) {
+                sketch.Add(short_id);
+            }
+        }
+        return sketch;
+    }
+
+public:
+
+    ReconciliationState(bool requestor, bool responder, bool flood_to, uint64_t k0, uint64_t k1) :
+        m_requestor(requestor), m_responder(responder), m_flood_to(flood_to),
+        m_k0(k0), m_k1(k1), m_local_q(DEFAULT_RECON_Q) {}
+
+    bool IsChosenForFlooding() const
+    {
+        return m_flood_to;
+    }
+
+    bool IsRequestor() const
+    {
+        return m_requestor;
+    }
+
+    bool IsResponder() const
```

Comment:
I think those 4 functions (with `GetLocalSetSize`) should be public. The remaining ones aren't used in`net_processing` and can be accessed by `TxReconciliationTracker` or its implementation by making it a friend of `ReconciliationState` ?